### PR TITLE
Add Unicode pattern table generator

### PR DESCRIPTION
**Pull request by Achie72**
The script uses WebKit algorithm to generate new tables
directly from the UCD files.

Signed-off-by: Bela Toth tbela@inf.u-szeged.hu

### CODE REVIEW

File: tools/genereate_unicode_pattern_tables.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+echo -e "\nReformat for Escargot"
+sed -i 's/makeUnique/std::make_unique/g' $PATTERN_FILE_PATH
+grep -v "generated" $PATTERN_FILE_PATH > temp_file; mv temp_file $PATTERN_FILE_PATH
+
+perl -0777 -i -p -e 's/,\n        CharacterClassWidths::HasBothBMPAndNonBMP\);/\);\n    characterClass->m_hasNonBMPCharacters = true;/g' $PATTERN_FILE_PATH
+perl -0777 -i -p -e 's/,\n        CharacterClassWidths::HasBMPChars\);/\);\n    characterClass->m_hasNonBMPCharacters = false;/g' $PATTERN_FILE_PATH
+perl -0777 -i -p -e 's/,\n        CharacterClassWidths::HasNonBMPChars\);/\);\n    characterClass->m_hasNonBMPCharacters = true;/g' $PATTERN_FILE_PATH
+
+#Cleanup
+sed -i 's/\s*$//g' $PATTERN_FILE_PATH
```
Comment (clover2123) [inline tools/genereate_unicode_pattern_tables.sh:54]: Would you please move this command line above the cleanup process?
It seems that this command does not relate to cleanup.
Other looks great! :)

Comment (Achie72) [reply tools/genereate_unicode_pattern_tables.sh:54]: Sure thing, let me re-push the code.

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by ksh8281 — APPROVED** [review]
What a good PR!

**Pull request merged**